src/cpu_shader/flatten.rs

```diff
@@ -258,127 +260,109 @@ fn flatten_euler(
-                        let k_peak = k0 - k1 * b / a;
-                        let integrand_peak = (k_peak * (k_peak * dist_scaled + 1.0)).abs().sqrt();
-                        let scaled_int = integral * integrand_peak / a;
-                        let n_frac = scaled_int;
-                        (n_frac, EspcRobust::Normal)
-                    };
-                    let n = (n_frac * scale_multiplier).ceil().max(1.0);
-
-                    // Flatten line segments
-                    log!("@@@   loop2: lines: {n}");
-                    // TODO: make all computation above robust and uncomment this assertion
-                    //assert!(!n.is_nan());
-                    if n.is_nan() {
-                        // Skip the segment if `n` is NaN. This is for debugging purposes only
-                        log!("@@@   NaN: parameters:\n  es: {:#?}\n  k0: {k0}, k1: {k1}\n  dist_scaled: {dist_scaled}\n  es_scale: {es_scale}\n  a: {a}\n  b: {b}\n  int0: {int0}, int1: {int1}, integral: {integral}\n  k_peak: {k_peak}\n  integrand_peak: {integrand_peak}\n  scaled_int: {scaled_int}\n  n_frac:  {n_frac}", es);
-                    } else {
-                        for i in 0..n as usize {
-                            let lp1 = if i == n as usize - 1 && t1 == 1.0 {
-                                t_end
-                            } else {
-                                let t = (i + 1) as f32 / n;
-                                let s = match robust {
-                                    EspcRobust::LowK1 => t,
-                                    // Note opportunities to minimize divergence
-                                    EspcRobust::LowDist => {
-                                        let c = (integral * t + int0).cbrt();
-                                        let inv = c * c.abs();
-                                        (inv - b) / a
-                                    }
-                                    EspcRobust::Normal => {
-                                        let inv = espc_int_inv_approx(integral * t + int0);
-                                        (inv - b) / a
-                                    }
-                                };
-                                es.eval_with_offset(s, offset)
-                            };
-                            let l0 = if offset >= 0. { lp0 } else { lp1 };
-                            let l1 = if offset >= 0. { lp1 } else { lp0 };
-                            output_line_with_transform(
-                                path_ix, l0, l1, &transform, line_ix, lines, bbox,
-                            );
-                            lp0 = lp1;
+        }
+        let actual_dt = t1 - last_t;
+        let cubic_params =
+            CubicParams::from_points_derivs(this_p0, this_p1, this_q0, this_q1, actual_dt);
+        log!("@@@   loop2: sub:{:?}, {:?} t0: {t0}, t1: {t1}, dt: {dt}, est_err: {est_err}, err: {err}", subcubic, cubic_params);
```

Comment:
Feel free to leave these debug logs in, though it might be worth changing the "loop1" and "loop2" tags given that there is a single loop now?